### PR TITLE
Fixed `PaginatedDataTable` not using `dataRowMinHeight` and `dataRowMaxHeight` from Theme

### LINKED ISSUE #133633: PaginatedDataTable widget ignoring dataTableTheme's dataRowMinHeight and dataRowMaxHeight
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Create new Flutter project
2. Add Theme to your MaterialApp widget and ensure the theme includes configuration for `dataTableTheme` and within that configuration have assigned a value to `dataRowMinHeight` and `dataRowMaxHeight`
3. Add a `PaginatedDataTable`
4. The `dataRowMinHeight` and `dataRowMaxHeight` values from your theme will have been ignored

### Expected results

The `PaginatedDataTable` will respect and adhere to the values assigned to the app theme's `dataTableTheme`.

### Actual results

Instead of making use of the theme values, it is using the value of `kMinInteractiveDimension` instead.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'dart:math';

import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Bug Report',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        dataTableTheme: const DataTableThemeData(
          dataRowMinHeight: 70.0,
          dataRowMaxHeight: 80.0,
        ),
        useMaterial3: true,
      ),
      home: Scaffold(
        body: SafeArea(
          child: Center(
            child: PaginatedDataTable(
              // dataRowMinHeight: 70.0,
              // dataRowMaxHeight: 80.0,
              source: MyDataSource(),
              header: Text('Table 1'),
              rowsPerPage: 5,
              columns: const [
                DataColumn(label: Flexible(child: Text('ID'))),
                DataColumn(label: Flexible(child: Text('Title'))),
                DataColumn(label: Flexible(child: Text('Cost'))),
              ],
            ),
          ),
        ), // This trailing comma makes auto-formatting nicer for build methods.
      ),
    );
  }
}

class MyDataSource extends DataTableSource {
  final List<Map<String, dynamic>> _data = List.generate(
    200,
    (index) {
      return {
        "id": '$index',
        "title": "Item $index",
        "cost": '${Random().nextInt(10000)}',
      };
    },
  );

  @override
  DataRow? getRow(int index) {
    final data = _data[index];
    return DataRow(
      cells: [
        DataCell(Text(data['id'])),
        DataCell(Text(data['title'])),
        DataCell(Text(data['cost'])),
      ],
    );
  }

  @override
  bool get isRowCountApproximate => false;

  @override
  int get rowCount => _data.length;

  @override
  int get selectedRowCount => 0;
}

```
</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.0, on macOS 13.5.1 22G90 darwin-arm64, locale en-GB)
    • Flutter version 3.13.0 on channel stable at /Users/mos/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision efbf63d9c6 (2 weeks ago), 2023-08-15 21:05:06 -0500
    • Engine revision 1ac611c64e
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/mos/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Users/mos/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13F17a
    ✗ Flutter requires Xcode 14 or higher.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Users/mos/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2)
    • IntelliJ at /Users/mos/Applications/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 75.1.4
    • Dart plugin version 232.8660.129

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.5.1 22G90 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 116.0.5845.110

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by MarkOSullivan94**
`PaginatedDataTable` will now make use of `dataRowMinHeight` and `dataRowMaxHeight` from the Theme

*List which issues are fixed by this PR. You must list at least one issue.*

Resolves #133633

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by TahaTesser** [comment]
@MarkOSullivan94 
The fix looks good.  `dataRowMinHeight` & `dataRowMaxHeight` should be  nullable & shouldn't default to `kMinInteractiveDimension`  and so the inner `DataTable` can be themed (while `DataTable` already defaults to the same value so the docs are still accurate)

This requires a test in the `paginated_data_table_test.dart` class to land.

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 1]
```diff
@@ -192,13 +192,13 @@ class PaginatedDataTable extends StatefulWidget {
   ///
   /// This value is optional and defaults to [kMinInteractiveDimension] if not
   /// specified.
-  final double dataRowMinHeight;
+  final double? dataRowMinHeight;
 
   /// The maximum height of each row (excluding the row that contains column headings).
   ///
   /// This value is optional and defaults to kMinInteractiveDimension if not
```
Comment (TahaTesser) [inline packages/flutter/lib/src/material/paginated_data_table.dart:199]:
```suggestion
  /// This value is optional and defaults to [kMinInteractiveDimension] if not
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 2]
```diff
@@ -1022,6 +1022,39 @@ void main() {
     await binding.setSurfaceSize(originalSize);
   });
 
+  testWidgets('dataRowMinHeight & dataRowMaxHeight if not set will use DataTableTheme', (WidgetTester tester) async {
+    final Size originalSize = binding.renderView.size;
+    await binding.setSurfaceSize(const Size(800, 800));
```
Comment (TahaTesser) [inline packages/flutter/test/material/paginated_data_table_test.dart:1027]:
```suggestion
    final Size originalSize = binding.renderView.size;
    addTearDown(() => binding.setSurfaceSize(null));    
    await binding.setSurfaceSize(const Size(800, 800));
```

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 3]
```diff
@@ -1022,6 +1022,39 @@ void main() {
...
+            DataColumn(label: Text('Name')),
+            DataColumn(label: Text('Calories'), numeric: true),
+            DataColumn(label: Text('Generation')),
+          ],
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildTable());
+    expect(tester.getSize(find.widgetWithText(Container, 'Frozen yogurt (0)').first).height, minMaxDataRowHeight);
```
Comment (TahaTesser) [inline packages/flutter/test/material/paginated_data_table_test.dart:1052]:
```suggestion
    await tester.pumpWidget(MaterialApp(
      theme: ThemeData(
        dataTableTheme: const DataTableThemeData(
          dataRowMinHeight: minMaxDataRowHeight,
          dataRowMaxHeight: minMaxDataRowHeight,
        ),
      ),
      home: PaginatedDataTable(
        header: const Text('Test table'),
        source: TestDataSource(allowSelection: true),
        columns: const <DataColumn>[
          DataColumn(label: Text('Name')),
          DataColumn(label: Text('Calories'), numeric: true),
          DataColumn(label: Text('Generation')),
        ],
      ),
    ));

    final Container rowContainer = tester.widget<Container>(find.descendant(
      of: find.byType(Table),
      matching: find.byType(Container),
    ).last);
    expect(rowContainer.constraints?.minHeight, minMaxDataRowHeight);
    expect(rowContainer.constraints?.maxHeight, minMaxDataRowHeight);
```

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 4]
```diff
@@ -1022,6 +1022,39 @@ void main() {
...
+
+    // Reset the surface size.
+    await binding.setSurfaceSize(originalSize);
```
Comment (TahaTesser) [inline packages/flutter/test/material/paginated_data_table_test.dart:1055]:
```suggestion
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/paginated_data_table_test.dart (hunk 2 above)
Comment (MarkOSullivan94) [reply packages/flutter/test/material/paginated_data_table_test.dart]: Thank you @TahaTesser

File: packages/flutter/test/material/paginated_data_table_test.dart (hunk 3 above)
Comment (MarkOSullivan94) [reply packages/flutter/test/material/paginated_data_table_test.dart:1053]: I agree this is better, thank you for the recommendation

File: packages/flutter/test/material/paginated_data_table_test.dart (hunk 4 above)
Comment (MarkOSullivan94) [reply packages/flutter/test/material/paginated_data_table_test.dart:1058]: Thank you

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 5]
```diff
@@ -1022,6 +1022,39 @@ void main() {
     await binding.setSurfaceSize(originalSize);
   });
 
+  testWidgets('dataRowMinHeight & dataRowMaxHeight if not set will use DataTableTheme', (WidgetTester tester) async {
+    final Size originalSize = binding.renderView.size;
```
Comment (TahaTesser) [inline packages/flutter/test/material/paginated_data_table_test.dart:1026]:
```suggestion
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 6]
```diff
@@ -1022,6 +1022,38 @@ void main() {
     await binding.setSurfaceSize(originalSize);
   });
 
+  testWidgets('dataRowMinHeight & dataRowMaxHeight if not set will use DataTableTheme', (WidgetTester tester) async {
+    addTearDown(() => binding.setSurfaceSize(null));
```
Comment (TahaTesser) [inline packages/flutter/test/material/paginated_data_table_test.dart:1026]:
```suggestion
    addTearDown(() => binding.setSurfaceSize(null));
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Review by TahaTesser — APPROVED** [review]
LGTM

**Comment by TahaTesser** [comment]
This will need a second review from @HansMuller before we land.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**